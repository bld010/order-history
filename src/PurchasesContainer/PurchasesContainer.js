import React from 'react';
import './PurchasesContainer.css';
import Purchase from '../Purchase/Purchase';

const PurchasesContainer = ({purchases, deletePurchase}) =>  {


  let purchaseList = purchases.map(purchase => {
    return <Purchase 
    key={purchase.id} 
    deletePurchase={deletePurchase}
    purchase={purchase} />
  })

  return(
    <>
    {purchaseList}
    </>
  )
}

export default PurchasesContainer; 