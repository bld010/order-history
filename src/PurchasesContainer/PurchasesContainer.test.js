import React from 'react';
import { shallow } from 'enzyme'
import PurchasesContainer from './PurchasesContainer';

describe('PurchasesContainer', () => {
  it('should match the snapshot', () => {
    const mockPurchases = [
      { 
        id: 1, 
        name: 'iPhone', 
        description: 'Expensive phone',
        img: 'www.apple.com/iphone.bmp',
        price: 900
       },
        { 
        id: 1, 
        name: 'iPhone 12', 
        description: 'More Expensive phone',
        img: 'www.apple.com/iphone.bmp',
        price: 92300
       }
    ]

    const wrapper = shallow(<PurchasesContainer
      purchases={mockPurchases}
      />)

    expect(wrapper).toMatchSnapshot();
  })


})