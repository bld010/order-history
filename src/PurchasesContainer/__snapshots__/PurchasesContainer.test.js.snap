// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PurchasesContainer should match the snapshot 1`] = `
<Fragment>
  <Purchase
    key="1"
    purchase={
      Object {
        "description": "Expensive phone",
        "id": 1,
        "img": "www.apple.com/iphone.bmp",
        "name": "iPhone",
        "price": 900,
      }
    }
  />
  <Purchase
    key="1"
    purchase={
      Object {
        "description": "More Expensive phone",
        "id": 1,
        "img": "www.apple.com/iphone.bmp",
        "name": "iPhone 12",
        "price": 92300,
      }
    }
  />
</Fragment>
`;
